design-proposals/klip-53-pull-queries-on-streams.md

```diff
@@ -27,17 +28,26 @@ pull-type queries.
 
 ## What is in scope
 
-* Queries should be supported on STREAM objects without “EMIT CHANGES” at the end
-* Pull queries should start at the beginning of the stream
-* Pull queries should terminate at the (current) end of the stream
+* Queries should be supported on STREAM objects without “EMIT CHANGES” at the end.
+* Pull queries should start at the beginning of the stream.
+* Pull queries should terminate. Streams do not have an "end" by definition, so we need to define
+  when the query terminates. When you issue the query, ksqlDB will scan over all the
+  data that is in the history of the stream at that moment and then terminate without
+  waiting for more data.
 * Pull queries on streams will support the same range of query operations as they
   do on tables. In other words, you will be able to filter the stream using a WHERE clause.
 
 ## What is not in scope
 
-* Pull queries on streams will not currently support grouping/aggregations.
+* Pull queries on streams will not currently support grouping/aggregations or joins.
   This would be useful but is left as future work.
-  Note that aggregations are also not currently supported on table pull queries.
+  Note that neither aggregations nor joins are currently supported on table pull queries.
+* If old records have already been dropped from the stream due to retention
+  settings, they will not be included in the results.
```

Comment:
Think we can omit this bullet point. Seems to be obvious?

Reply:
Thanks :). I also felt like it might be a little too obvious, but I went ahead and specified it in case people don't immediately put two and two together regarding the interaction of retention and their stream queries.